numba/ir_utils.py

```diff
@@ -902,6 +902,54 @@ def simplify_CFG(blocks):
                 label_map[next_node] = node
     return
 
+arr_math = ['min', 'max', 'sum', 'prod', 'mean', 'var', 'std',
+             'cumsum', 'cumprod', 'argmin', 'argmax', 'argsort',
+             'nonzero', 'ravel']
+
+def canonicalize_array_math(blocks, typemap, calltypes, typingctx):
+    # save array arg to call
+    # call_varname -> array
+    saved_arr_arg = {}
+    topo_order = find_topo_order(blocks)
+    for label in topo_order:
+        block = blocks[label]
+        new_body = []
+        for stmt in block.body:
+            if isinstance(stmt, ir.Assign) and isinstance(stmt.value, ir.Expr):
+                lhs = stmt.target.name
+                rhs = stmt.value
+                if (rhs.op=='getattr' and rhs.attr in arr_math
+                    and isinstance(
+                    typemap[rhs.value.name], types.npytypes.Array)):
+                    rhs = stmt.value
+                    arr = rhs.value
+                    saved_arr_arg[lhs] = arr
+                    scope = arr.scope
+                    loc = arr.loc
+                    # g_np_var = Global(numpy)
```

Comment:
Ah, apologies, I was reading `Global` as a function and this as dead code, this is fine :)